klippy/extras/bed_mesh.py

```diff
@@ -484,6 +525,71 @@ def print_probed_positions(self, print_func):
             print_func(msg)
         else:
             print_func("bed_mesh: bed has not been probed")
+
+    # Look up z position according to the active mesh and
+    # compare with probed height
+    def _calculate_delta_z(self, pos, z_offset):
+        z_in_mesh = self.bedmesh.z_mesh.calc_z(*pos[0:2]) + z_offset
+        delta_z = pos[2] - z_in_mesh;
+        return [pos[0], pos[1], delta_z]
+
+    def tilt_probe_finalize(self, offsets, positions):
+        x_offset, y_offset, z_offset = offsets
+        self.gcode.respond_info("offsets: %f %f %f" % tuple(offsets));
+
+        if self.relative_reference_index is not None:
+            rri = self.relative_reference_index
+            z_offset = self.bedmesh.z_mesh.calc_z(*self.points[rri][0:2])
```

Comment:
> Arksine: Thinking further about the z offset, could we just take an average of the tilt points, then offset them using that value? It seems that the offset from the mesh isn't really important, what we are interested in is a change in the delta among the tilt points.

Thanks for the thoughtful comments. 

I was a bit thrown off in my thinking by the `relative_reference_index` issue, as it seemed to suggest that the bed mesh system had some role in establishing the zero level for the Z axis as a whole. But reading more closely through the code, that appears to be an entirely separate mechanism and is controlled by the axis settings or the probe offset.

I believe you're right about offsets being irrelevant for meshes, at least as far as I can determine with my partial understanding of the code. Either the user will specify a fade target, which will act as a global offset, or the code will calculate an average Z value and use that as the offset when the mesh is activated. Offsets applied to incoming meshes are therefore irrelevant except to the extent that they might alter the value the user needs to specify as a fade target, if that feature is in use.

A corollary is that `relative_reference_index` doesn't really seem to have a functional role anywhere in the current codebase. In `probe_finalize` it specifies an alternative offset applied to incoming probe values (in lieu of the probe's own Z offset), but since the value is constant it has no real effect, as you outlined (and neither would the probe's Z offset). `tilt_probe_finalize` is just matching the general flow of `probe_finalize`, but the adjustment is equally unnecessary in both places, is it not?

I'm wondering if, in addition to normalizing probe matrices to an average vale of zero, it might be possible to just deprecate `relative_reference_index` and the code that uses it.